llama-index-core/llama_index/core/schema.py

```diff
@@ -389,7 +431,123 @@ def as_related_node_info(self) -> RelatedNodeInfo:
         )
 
 
+class MediaResource(BaseModel):
+    """A container class for media content.
+
+    This class represents a generic media resource that can be stored and accessed
+    in multiple ways - as raw bytes, on the filesystem, or via URL. It also supports
+    storing vector embeddings for the media content.
+
+    Attributes:
+        embeddings: Dense vector representation of this resource for embedding-based search/retrieval
+        sparse: Sparse vector representation of this resource
+        data: Raw binary data of the media content
+        mimetype: The MIME type indicating the format/type of the media content
+        path: Local filesystem path where the media content can be accessed
+        url: URL where the media content can be accessed remotely
+    """
+
+    embeddings: list[float] | None = Field(
+        default=None, description="Vector representation of this resource."
+    )
+    sparse: list[float] | None = Field(
+        default=None, description="Sparse vector representation of this resource."
+    )
+    data: bytes | None = Field(
+        default=None, exclude=True, description="Binary data of this resource."
+    )
+    mimetype: str | None = Field(
```

Comment:
we recently added a mime type library to core, I wonder if the default can be using that to guess the mime type if not directly set?

```
import filetype

# Decode the base64 string
decoded_data = base64.b64decode(base64_string)

# Use filetype to guess the MIME type
kind = filetype.guess(decoded_data)
```

Reply:
Done! Please have a look if the model validator looks like a viable solution

Reply:
seems pretty good to me!